pkg/objstore/swift/swift.go

```diff
@@ -8,75 +8,191 @@ import (
 	"context"
 	"fmt"
 	"io"
+	"io/ioutil"
 	"os"
+	"strconv"
 	"strings"
 	"testing"
+	"time"
+
+	"github.com/go-kit/kit/log/level"
+
+	"github.com/thanos-io/thanos/pkg/runutil"
 
 	"github.com/go-kit/kit/log"
-	"github.com/gophercloud/gophercloud"
-	"github.com/gophercloud/gophercloud/openstack"
-	"github.com/gophercloud/gophercloud/openstack/objectstorage/v1/containers"
-	"github.com/gophercloud/gophercloud/openstack/objectstorage/v1/objects"
-	"github.com/gophercloud/gophercloud/pagination"
+	"github.com/ncw/swift"
 	"github.com/pkg/errors"
 	"github.com/thanos-io/thanos/pkg/objstore"
 	"gopkg.in/yaml.v2"
 )
 
 // DirDelim is the delimiter used to model a directory structure in an object store bucket.
-const DirDelim = "/"
-
-type SwiftConfig struct {
-	AuthUrl           string `yaml:"auth_url"`
-	Username          string `yaml:"username"`
-	UserDomainName    string `yaml:"user_domain_name"`
-	UserDomainID      string `yaml:"user_domain_id"`
-	UserId            string `yaml:"user_id"`
-	Password          string `yaml:"password"`
-	DomainId          string `yaml:"domain_id"`
-	DomainName        string `yaml:"domain_name"`
-	ProjectID         string `yaml:"project_id"`
-	ProjectName       string `yaml:"project_name"`
-	ProjectDomainID   string `yaml:"project_domain_id"`
-	ProjectDomainName string `yaml:"project_domain_name"`
-	RegionName        string `yaml:"region_name"`
-	ContainerName     string `yaml:"container_name"`
+const DirDelim = '/'
+
+var DefaultConfig = Config{
```

Comment:
Maybe it would be worth adding `AuthVersion` here with the value `2`? AFAIK there are only two auth versions available and `2` is the newer one, so adding this default value here would mean that all of the Swift users won't need to add it.

Reply:
Hm, Actually there are 3 versions, see for example http://doc.isilon.com/ECS/3.1/DataAccessGuide/ecs_c_oapi_swift_authentication_introduction.html

But we could do better with the default value. The library docs states: `Set to 1, 2 or 3 or leave at 0 for autodetect` so `0` might be the way WDYT?

Reply:
Changed to `0` now

Reply:
Makes sense, thank you!